ports/stm32/rfcore.c

```diff
@@ -413,9 +411,10 @@ STATIC size_t tl_process_msg(volatile tl_list_node_t *head, unsigned int ch, par
     bool added_to_free_queue = false;
     size_t len = 0;
     while (cur != head) {
+        volatile tl_list_node_t *next = tl_list_unlink(cur);
+        
```

Comment:
This change (moving the line up) is what solved my HardFault... 
In the processing of `tl_parse_hci_msg()` below a new message might be sent which changes the cur / head of `ipcc_mem_ble_evt_queue`
"snapshotting" the cur / next here before processing preserves them. Indeed removing it from the linked list before processing perhaps makes sense? Or might be dangerous, I'm not entirely certain...